### PR TITLE
[WIP] Offline User Inventory + Equipable

### PR DESCRIPTION
**Pull request by Faithcaio**
*SpongeAPI* | [SpongeCommon](https://github.com/SpongePowered/SpongeCommon/pull/1401)

Add support for User Inventories.
Also some missing parts of equipable.

### CODE REVIEW

File: src/main/java/org/spongepowered/api/item/inventory/equipment/EquipmentTypeHeld.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.inventory.equipment;
+
+/**
+ * An equipment item which can be held. E.g. main or offhand
+ */
+public interface EquipmentTypeHeld extends EquipmentType {
```
Comment (kashike) [inline src/main/java/org/spongepowered/api/item/inventory/equipment/EquipmentTypeHeld.java]: `HeldEquipmentType`

**Review by kashike — CHANGES_REQUESTED** [review]

File: src/main/java/org/spongepowered/api/item/inventory/equipment/EquipmentTypeHeld.java (hunk 1 above)
Comment (Faithcaio) [reply src/main/java/org/spongepowered/api/item/inventory/equipment/EquipmentTypeHeld.java]: should I also rename EquipmentTypeWorn then?

Comment (Zidane) [inline src/main/java/org/spongepowered/api/item/inventory/equipment/EquipmentTypeHeld.java]: Yes

File: src/main/java/org/spongepowered/api/item/inventory/equipment/EquipmentTypes.java [hunk 2]
```diff
@@ -45,26 +42,27 @@
...
     /**
-     * Armor types, HEADWEAR, CHESTPLATE, LEGGINGS and BOOTS should subclass
-     * this.
+     * Any held types like MAINHAND and OFFHAND
      */
-    public static final EquipmentTypeWorn WORN = DummyObjectProvider.createFor(EquipmentTypeWorn.class, "WORN");
+    public static final EquipmentTypeHeld HELD = DummyObjectProvider.createFor(EquipmentTypeHeld.class, "HELD");
 
-    // SORTFIELDS:OFF
+    public static final EquipmentTypeHeld MAINHAND = DummyObjectProvider.createFor(EquipmentTypeHeld.class, "MAINHAND");
```
Comment (Cybermaxke) [inline src/main/java/org/spongepowered/api/item/inventory/equipment/EquipmentTypes.java]: `MAIN_HAND` and `OFF_HAND`?

File: src/main/java/org/spongepowered/api/item/inventory/equipment/EquipmentTypeHeld.java (hunk 1 above)
Comment (kashike) [reply src/main/java/org/spongepowered/api/item/inventory/equipment/EquipmentTypeHeld.java]: yes

File: src/main/java/org/spongepowered/api/item/inventory/entity/UserInventory.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+public interface UserInventory<T extends User> extends CarriedInventory<T> {
+
+    /**
+     * Gets the hotbar inventory.
+     *
+     * @return The hotbar
+     */
+    Hotbar getHotbar();
+
+    // TODO getMainPlayerInventory which includes the hotbar
```
Comment (Zidane) [inline src/main/java/org/spongepowered/api/item/inventory/entity/UserInventory.java]: TODO?

Comment (Cybermaxke) [reply src/main/java/org/spongepowered/api/item/inventory/entity/UserInventory.java]: When I see `getMain`, I would think it contains the complete bottom inventory, **with** the hotbar. This is also how I implemented in Lantern. If without the hotbar is the way to go, I would also clarify this in the javadocs.

Comment (Faithcaio) [reply src/main/java/org/spongepowered/api/item/inventory/entity/UserInventory.java]: Currently it does not contain the Hotbar. 
I want to change that.

Comment (Zidane) [reply src/main/java/org/spongepowered/api/item/inventory/entity/UserInventory.java]: @Faithcaio 

See about making that change as it makes sense so long as we get a distinct method for the view of the Inventory WITHOUT the Hotbar.

File: src/main/java/org/spongepowered/api/item/inventory/entity/UserInventory.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import org.spongepowered.api.entity.living.player.User;
+import org.spongepowered.api.item.inventory.Slot;
+import org.spongepowered.api.item.inventory.crafting.CraftingInventory;
+import org.spongepowered.api.item.inventory.equipment.EquipmentInventory;
+import org.spongepowered.api.item.inventory.type.CarriedInventory;
+import org.spongepowered.api.item.inventory.type.GridInventory;
+
+/**
+ * Represents the inventory of a {@link User}.
+ * It consists of
```
Comment (Zidane) [inline src/main/java/org/spongepowered/api/item/inventory/entity/UserInventory.java]: Docs fail

**Pull request merged**